### PR TITLE
Documentation on gpu float16 ops.

### PR DESCRIPTION
**Pull request by abergeron**
(and some cleanup).

### CODE REVIEW

File: theano/gpuarray/fp16_help.py [hunk 1]
```diff
@@ -2,20 +2,40 @@
...
         return dtype
 
 
 def load_w(dtype):
+    """
+    Return the function name to load data.
+
+    This should be used like this::
+
+        code = '%(load_f)s(ival)' % (load_w(input_type),)
```
Comment (nouiz) [inline theano/gpuarray/fp16_help.py]: this is not valid python code. I think you mean:

    code = '%s(ival)' % (load_w(input_type),)

File: theano/gpuarray/fp16_help.py [hunk 2]
```diff
@@ -2,20 +2,40 @@
...
         return ''
 
 
 def write_w(dtype):
+    """
+    Return the function name to write data.
+
+    This should be used like this::
+
+        code = 'res = %(write_f)s(oval)' % (write_w(output_type),)
```
Comment (nouiz) [inline theano/gpuarray/fp16_help.py]: idem

File: doc/extending/extending_theano_gpu.txt [hunk 3]
```diff
@@ -198,6 +198,47 @@ by :function:`load_w`. Similarly writes should be wrapped in the
...
+
+    type_x = dtype_to_ctype(x.dtype)
+    type_y = dtype_to_ctype(y.dtype)
+    work_x = dtype_to_ctype(work_dtype(x.dtype))
+    load_x = load_w(x.dtype)
+    write_y = write_w(y.dtype)
+    """
+    KERNEL void k(const ga_size n, %(type_x)s *x, %(type_y)s *y) {
+        ga_size i = GID_0 * LDIM_0 + LID_0;
+        %(work_x) z = %(load_w)(x[i]);
```
Comment (nouiz) [inline doc/extending/extending_theano_gpu.txt]: load_w -> load_x

File: doc/extending/extending_theano_gpu.txt [hunk 4]
```diff
@@ -198,6 +198,47 @@ by :function:`load_w`. Similarly writes should be wrapped in the
...
+        z += 1;
+        y[i] = %(write_w)(z);
```
Comment (nouiz) [inline doc/extending/extending_theano_gpu.txt]: write_w -> write_y

File: doc/extending/extending_theano_gpu.txt [hunk 5]
```diff
@@ -198,6 +198,47 @@ by :function:`load_w`. Similarly writes should be wrapped in the
...
+Once you have converted your kernels for float16 support you need to
+tag your op with ``_f16_ok = True`` so that the linker will accept to
+generate C code on float16 types.  This is done by inserting it as a
+class property like this::
+
+    class SomeOp(Op):
+        _f16_ok = True
+
+If this attribute is not present or is False, the linker will print a
+message saying that it's refusing to use C code for float16.
```
Comment (nouiz) [inline doc/extending/extending_theano_gpu.txt]: for float16 for this op.

File: doc/extending/extending_theano_gpu.txt [hunk 6]
```diff
@@ -156,25 +156,92 @@ in most cases in your :meth:`c_code` method.  This is done by using
...
+generate C code on float16 types.  This is done by inserting it as a
+class property like this::
+
+    class SomeOp(Op):
+        _f16_ok = True
+
+If this attribute is not present or is False, the linker will print a
+message saying that it's refusing to use C code for float16 for the
+op.
+
```
Comment (nouiz) [inline doc/extending/extending_theano_gpu.txt]: I think this warant a complete example (see section bellow) that would be a doctest, so we are sure the doc continue to work.

Comment (abergeron) [inline doc/extending/extending_theano_gpu.txt]: The GpuEye example does have the `_f16_ok` tag.  Unless you mean something else?

Comment (nouiz) [inline doc/extending/extending_theano_gpu.txt]: that should do it, so I merge. thanks.

**Pull request merged**